src/models/treemodel.h

```diff
@@ -281,6 +278,46 @@ class CostTreeModel : public TreeModel<decltype(Results::root), ModelImpl>
     Results m_results;
 };
 
+template<typename Results, typename ModelImpl>
+class CostProxy : public QSortFilterProxyModel
+{
+public:
+    explicit CostProxy(QObject* parent = nullptr)
+    {
+        setRecursiveFilteringEnabled(true);
+        setDynamicSortFilter(true);
+    }
+
+protected:
+    bool filterAcceptsRow(int source_row, const QModelIndex& parent) const override
+    {
+        if (parent.isValid())
```

Comment:
this filters out everything but the top level entries which sounds wrong to me?

Reply:
yeah